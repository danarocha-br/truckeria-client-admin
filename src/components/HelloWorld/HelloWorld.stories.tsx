import { Story, Meta } from '@storybook/react/types-6-0';

import Heading from '.';

export default {
  title: 'Example/Heading',
  component: Heading,
} as Meta;

export const Default: Story = (args) => <Heading {...args} />;
