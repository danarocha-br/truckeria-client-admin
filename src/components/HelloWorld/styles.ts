import styled, { css } from 'styled-components';

export const Heading = styled.h1`
  ${({ theme }) => css`
    color: ${theme.text.main};
  `}
`;
