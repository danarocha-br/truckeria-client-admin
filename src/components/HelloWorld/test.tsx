import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/helpers';
import HelloWorld from '.';

describe('<HelloWorld />', () => {
  it('should render the heading on HelloWorld component', () => {
    const { container } = renderWithTheme(<HelloWorld />);

    expect(
      screen.getByRole('heading', { name: /Hello World/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
