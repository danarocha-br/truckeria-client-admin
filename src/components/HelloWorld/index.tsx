import { Heading } from './styles';

// export interface TestProps {
//   /**
//    * Is this the principal call to action on the page?
//    */
//   primary?: boolean;
//   /**
//    * What background color to use
//    */
//   backgroundColor?: string;
//   /**
//    * How large should the button be?
//    */
//   size?: 'small' | 'medium' | 'large';
//   /**
//    * Button contents
//    */
//   label: string;
//   /**
//    * Optional click handler
//    */
//   onClick?: () => void;
// }

const Test: React.FC = () => {
  return <Heading>Hello World</Heading>;
};

export default Test;
