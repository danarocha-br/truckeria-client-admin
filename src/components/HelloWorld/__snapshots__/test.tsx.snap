// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<HelloWorld /> should render the heading on HelloWorld component 1`
] = `
.c0 {
  color: var(--color-brand);
}

<h1
  class="c0"
>
  Hello World
</h1>
`;
